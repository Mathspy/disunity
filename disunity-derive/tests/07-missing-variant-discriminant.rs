use disunity_derive::Variant;

#[derive(Variant)]
enum Class {
    // Unknown is special and doesn't need a discriminant
    Unknown(i32),
    #[disunity(discriminant = 1)]
    GameObject {
        field: i32,
    },
    Transform,
    Magic(i32),
}

fn main() {}
