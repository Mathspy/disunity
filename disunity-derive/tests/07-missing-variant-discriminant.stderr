error: variant without a #[disunity(discriminant = N)] attribute
  --> tests/07-missing-variant-discriminant.rs:11:5
   |
11 |     Transform,
   |     ^^^^^^^^^
